packages/mattermost-redux/src/types/posts.ts

```diff
@@ -27,12 +27,14 @@ export type PostType = 'system_add_remove' |
 'system_purpose_change' |
 'system_remove_from_channel';
 
-export type PostEmbedType = 'image' | 'link' | 'message_attachment' | 'opengraph';
+export type PostEmbedType = 'image' | 'link' | 'message_attachment' | 'opengraph' | 'permalink';
+
+export type AttachmentTextOverflowType = 'ellipsis';
 
 export type PostEmbed = {
     type: PostEmbedType;
     url: string;
-    data: Record<string, OpenGraphMetadata>;
+    data?: Record<string, OpenGraphMetadata> | PostPreviewMetadata;
```

Comment:
I was so confused about why it was setup like this but now I understand. I never knew there was a store in redux with metadata mapped to the url but I don't know why it's like this for the PostEmbed type. There are no instances where the `data` attribute is setup with metadata mapped to the URL. I'm actually going to update this type to:
```
export type PostEmbed = {
    type: PostEmbedType;
    url: string;
    data?: OpenGraphMetadata | PostPreviewMetadata;
}
```

Reply:
That does sound correct. I don't know why there's a Record in the current definition of PostEmbed, but I wouldn't be surprised if the current definition is incorrect since some of our Redux types are wonky.